### ISSUE TITLE
[Bug]: min_seeding_time not working

### ISSUE THREAD

**Issue by nothing2obvi** [StuffAnThings/qbit_manage#533]
### Describe the Bug

`min_seeding_time` doesn't seem to work on any of my torrents. For most, but not all torrents, once they're completed they don't seed. If they complete in 4 minutes but my `min_seeding_time` is set to 60 minutes, it has no effect. It won't seed. You can look specifically at my `public` share-limit.

I don't think the logs help.

Am I misunderstanding something?

### Config

```yaml
# This is an example configuration file that documents all the options.
# It will need to be modified for your specific use case.
# Please refer to the link below for more details on how to set up the configuration file
# https://github.com/StuffAnThings/qbit_manage/wiki/Config-Setup

commands:
  # The commands defined below will IGNORE any commands used in command line and docker env variables.
  dry_run: False
  cross_seed: False
  recheck: False
  cat_update: True
  tag_update: True
  rem_unregistered: True
  tag_tracker_error: False
  rem_orphaned: True
  #tag_nohardlinks: False
  share_limits: True
  skip_qb_version_check: False
  skip_cleanup: False

  tag_nohardlinks: False
qbt:
  # qBittorrent parameters
  host: ip:port
  user: username
  pass: password

settings:
  force_auto_tmm: False # Will force qBittorrent to enable Automatic Torrent Management for each torrent.
  #tracker_error_tag: iss # Will set the tag of any torrents that do not have a working tracker.
  nohardlinks_tag: noHL # Will set the tag of any torrents with no hardlinks.
  share_limits_tag: sl # Will add this tag when applying share limits to provide an easy way to filter torrents by share limit group/priority for each torrent
  share_limits_min_seeding_time_tag: nmst # Tag to be added to torrents that have not yet reached the minimum seeding time
  share_limits_min_num_seeds_tag: nmsn # Tag to be added to torrents that have not yet reached the minimum number of seeds
  share_limits_last_active_tag: nla # Tag to be added to torrents that have not yet reached the last active limit
  ignoreTags_OnUpdate: # When running tag-update function, it will update torrent tags for a given torrent even if the torrent has at least one or more of the tags defined here. Otherwise torrents will not be tagged if tags exist.
	- noHL
	- iss
	- cs
	- mst
	- msn
	- nla
  cross_seed_tag: cs # Will set the tag of any torrents that are added by cross-seed command
  cat_filter_completed: True # Filters for completed torrents only when running cat_update command
  share_limits_filter_completed: True # Filters for completed torrents only when running share_limits command
  tag_nohardlinks_filter_completed: True # Filters for completed torrents only when running tag_nohardlinks command
  force_retag_all: False # Forces all torrents to be updated on every run (may cause performance issues). This does not remove any previous tags assigned
directory:
  # Do not remove these
  # Cross-seed var: </your/path/here/>  # Output directory of cross-seed
  # root_dir var: </your/path/here/>  # Root downloads directory used to check for orphaned files, noHL, and RecycleBin.
  # <OPTIONAL> remote_dir var: </your/path/here/>  # Path of docker host mapping of root_dir.
  # remote_dir must be set if you're running qbit_manage locally and qBittorrent/cross_seed is in a docker
  # remote_dir should not be set if qbit_manage is running in a container
  # <OPTIONAL> recycle_bin var: </your/path/here/>   # Path of the RecycleBin folder. Default location is set to remote_dir/.RecycleBin
  # <OPTIONAL> torrents_dir var: </your/path/here/>  # Path of the your qbittorrent torrents directory. Required for `save_torrents` attribute in recyclebin
  # <OPTIONAL> orphaned_dir var: </your/path/here/>  # Path of the the Orphaned Data folder. This is similar to RecycleBin, but only for orphaned data.
  cross_seed: /home/username/downloads/qbittorrent/__cross-seed
  root_dir: /home/username/downloads/qbittorrent
  #remote_dir: "/mnt/user/data/torrents/"
  recycle_bin: /home/username/downloads/qbittorrent/__recycle-bin
  #torrents_dir: "/qbittorrent/data/BT_backup"
  orphaned_dir: /home/username/downloads/qbittorrent/__orphaned_data

  remote_dir: /home/username/downloads/qbittorrent/
  torrents_dir:
cat:
  # Category & Path Parameters
  # <Category Name> : <save_path>  # Path of your save directory.
  tv-sonarr: /home/username/downloads/qbittorrent/__tv-sonarr
  movies-radarr: /home/username/downloads/qbittorrent/__movies-radarr
  private-sonarr: /home/username/downloads/qbittorrent/__private-sonarr
  private-radarr: /home/username/downloads/qbittorrent/__private-radarr

cat_change:
  # This moves all the torrents from one category to another category. This executes on --cat-update
  # WARNING: if the paths are different and Default Torrent Management Mode is set to automatic the files could be moved !!!
  # <Old Category Name> : <New Category>
  #Radarr-HD.cross-seed: movies-hd
  #Radarr-UHD.cross-seed: movies-uhd
  #movies-hd.cross-seed: movies-hd
  #movies-uhd.cross-seed: movies-uhd

tracker:
  # Mandatory
  # Tag Parameters
  # <Tracker URL Keyword>:    # <MANDATORY> This is the keyword in the tracker url. You can define multiple tracker urls by splitting with `|` delimiter
  # <MANDATORY> Set tag name. Can be a list of tags or a single tag
  #   tag: <Tag Name>
  # <OPTIONAL> Set the category based on tracker URL. This category option takes priority over the category defined by save directory
  #   cat: <Category Name>
  # <OPTIONAL> Set this to the notifiarr react name. This is used to add indexer reactions to the notifications sent by Notifiarr
  #   notifiarr: <notifiarr indexer>
  tagnames:
	something:
	  - something
   another-thing:
     - another-thing
   again-another-thing:
     - again-another-thing
  # The "other" key is a special keyword and if defined will tag any other trackers that don't match the above trackers into this tag
  other:
	tag: other

nohardlinks:
  # Tag Movies/Series that are not hard linked outside the root directory
  # Mandatory to fill out directory parameter above to use this function (root_dir/remote_dir)
  # This variable should be set to your category name of your completed movies/completed series in qbit. Acceptable variable can be any category you would like to tag if there are no hardlinks found
  #- movies-completed-4k
  #- series-completed-4k
  #- movies-completed:
  # <OPTIONAL> exclude_tags var: Will exclude torrents with any of the following tags when searching through the category.
  #exclude_tags:
  #- Beyond-HD
  #- AnimeBytes
  #- MaM
  # Can have additional categories set with separate ratio/seeding times defined.
  #- series-completed:
  # <OPTIONAL> exclude_tags var: Will exclude torrents with any of the following tags when searching through the category.
  #exclude_tags:
  #- Beyond-HD
  #- BroadcasTheNet

share_limits:
  # Control how torrent share limits are set depending on the priority of your grouping
  # Each torrent will be matched with the share limit group with the highest priority that meets the group filter criteria.
  # Each torrent can only be matched with one share limit group
  # This variable is mandatory and is a text defining the name of your grouping. This can be any string you want
  private:
	priority: 1
	include_any_tags:
	  - something
	categories:
	  - private-sonarr
	  - private-radarr
	  - autobrr
	max_ratio: 999
	max_seeding_time: 20180
	min_seeding_time: 20170
	resume_torrent_after_change: True
	cleanup: False
  another-thing:
	priority: 2
	include_any_tags:
	  - another-thing
	max_ratio: 3
	max_seeding_time: 10100
	min_seeding_time: 10090
	resume_torrent_after_change: True
	cleanup: True
  again-another-thing:
	priority: 3
	include_any_tags:
	  - again-another-thing
	max_ratio: 2
	max_seeding_time: 10100
	min_seeding_time: 10090
	resume_torrent_after_change: True
	cleanup: False
  noHL:
	# <MANDATORY> priority: <int/float> # This is the priority of your grouping. The lower the number the higher the priority
	priority: 8
	# <OPTIONAL> include_all_tags: <list> # Filter the group based on one or more tags. Multiple include_all_tags are checked with an AND condition
	# All tags defined here must be present in the torrent for it to be included in this group
	include_all_tags:
	  - noHL
	# <OPTIONAL> include_any_tags: <list> # Filter the group based on one or more tags. Multiple include_any_tags are checked with an OR condition
	# Any tags defined here must be present in the torrent for it to be included in this group
	include_any_tags:
	  - noHL
	# <OPTIONAL> exclude_all_tags: <list> # Filter by excluding one or more tags. Multiple exclude_all_tags are checked with an AND condition
	# This is useful to combine with the category filter to exclude one or more tags from an entire category
	# All tags defined here must be present in the torrent for it to be excluded in this group
	#exclude_all_tags:
	#- Beyond-HD
	# <OPTIONAL> exclude_any_tags: <list> # Filter by excluding one or more tags. Multiple exclude_any_tags are checked with an OR condition
	# This is useful to combine with the category filter to exclude one or more tags from an entire category
	# Any tags defined here must be present in the torrent for it to be excluded in this group
	#exclude_any_tags:
	#- Beyond-HD
	# <OPTIONAL> categories: <list> # Filter by including one or more categories. Multiple categories are checked with an OR condition
	# Since one torrent can only be associated with a single category, multiple categories are checked with an OR condition
	#categories:
	#- RadarrComplete
	#- SonarrComplete
	# <OPTIONAL> max_ratio <float>: Will set the torrent Maximum share ratio until torrent is stopped from seeding/uploading and may be cleaned up / removed if the minimums have been met.
	# Will default to -1 (no limit) if not specified for the group.
	#max_ratio: 5.0
	# <OPTIONAL> max_seeding_time <int>: Will set the torrent Maximum seeding time (minutes) until torrent is stopped from seeding/uploading and may be cleaned up / removed if the minimums have been met.
	# Will default to -1 (no limit) if not specified for the group.
	#max_seeding_time: 129600
	# <OPTIONAL> min_seeding_time <int>: Will prevent torrent deletion by cleanup variable if torrent has not yet minimum seeding time (minutes).
	# This should only be set if you are using this in conjunction with max_seeding_time and max_ratio. If you are not setting a max_ratio, then use max_seeding_time instead.
	# If the torrent has not yet reached this minimum seeding time, it will change the share limits back to no limits and resume the torrent to continue seeding.
	# Will default to 0 if not specified for the group.
	#min_seeding_time: 43200
	# <OPTIONAL> last_active <int>: Will prevent torrent deletion by cleanup variable if torrent has been active within the last x minutes.
	# If the torrent has been active within the last x minutes, it will change the share limits back to no limits and resume the torrent to continue seeding.
	# Will default to 0 if not specified for the group.
	#last_active: 43200
	# <OPTIONAL> Limit Upload Speed <int>: Will limit the upload speed KiB/s (KiloBytes/second) (`-1` : No Limit)
	#limit_upload_speed: 0
	# <OPTIONAL> Enable Group Upload Speed <bool>: Upload speed limits are applied at the group level. This will take limit_upload_speed defined and divide it equally among the number of torrents in the group.
	#enable_group_upload_speed: False
	# <OPTIONAL> cleanup <bool>: WARNING!! Setting this as True Will remove and delete contents of any torrents that satisfies the share limits (max time OR max ratio)
	#cleanup: False
	# <OPTIONAL> resume_torrent_after_change <bool>: This variable will resume your torrent after changing share limits. Default is True
	#resume_torrent_after_change: True
	# <OPTIONAL> add_group_to_tag <bool>: This adds your grouping as a tag with a prefix defined in settings . Default is True
	# Example: A grouping defined as noHL will have a tag set to ~share_limit.noHL (if using the default prefix)
	#add_group_to_tag: True
	# <OPTIONAL> min_num_seeds <int>: Will prevent torrent deletion by cleanup variable if the number of seeds is less than the value set here.
	# If the torrent has less number of seeds than the min_num_seeds, the share limits will be changed back to no limits and resume the torrent to continue seeding.
	# Will default to 0 if not specified for the group.
	#min_num_seeds: 0
	cleanup: False
  public:
	priority: 9
	exclude_any_tags:
	  - something
	  - another-thing
	  - again-another-thing
	max_ratio: 1
	min_seeding_time: 60
	max_seeding_time: 70
	resume_torrent_after_change: True
	cleanup: False

recyclebin:
  # Recycle Bin method of deletion will move files into the recycle bin (Located in /root_dir/.RecycleBin) instead of directly deleting them in qbit
  # By default the Recycle Bin will be emptied on every run of the qbit_manage script if empty_after_x_days is defined.
  enabled: True
  # <OPTIONAL> empty_after_x_days var:
  # Will automatically remove all files and folders in recycle bin after x days. (Checks every script run)
  # If this variable is not defined it, the RecycleBin will never be emptied.
  # WARNING: Setting this variable to 0 will delete all files immediately upon script run!
  empty_after_x_days: 3
  # <OPTIONAL> save_torrents var:
  # If this option is set to True you MUST fill out the torrents_dir in the directory attribute.
  # This will save a copy of your .torrent and .fastresume file in the recycle bin before deleting it from qbittorrent
  save_torrents: False
  # <OPTIONAL> split_by_category var:
  # This will split the recycle bin folder by the save path defined in the `cat` attribute
  # and add the base folder name of the recycle bin that was defined in the `recycle_bin` sub-attribute under directory.
  split_by_category: True

orphaned:
  # Orphaned files are those in the root_dir download directory that are not referenced by any active torrents.
  # Will automatically remove all files and folders in orphaned data after x days. (Checks every script run)
  # If this variable is not defined it, the orphaned data will never be emptied.
  # WARNING: Setting this variable to 0 will delete all files immediately upon script run!
  empty_after_x_days: 60
  # File patterns that will not be considered orphaned files. Handy for generated files that aren't part of the torrent but belong with the torrent's files
  exclude_patterns:
	- "**/.DS_Store"
	- "**/Thumbs.db"
	- "**/@eaDir"
	- /data/torrents/temp/**
	- "**/*.!qB"
	- "**/*_unpackerred"
	- "**/**"
#apprise:
# Apprise integration with webhooks
# Leave Empty/Blank to disable
# Mandatory to fill out the url of your apprise API endpoint
#api_url: http://apprise-api:8000
# Mandatory to fill out the notification url/urls based on the notification services provided by apprise. https://github.com/caronc/apprise/wiki
#notify_url:

#notifiarr:
# Notifiarr integration with webhooks
# Leave Empty/Blank to disable
# Mandatory to fill out API Key
#apikey: ####################################
# <OPTIONAL> Set to a unique value (could be your username on notifiarr for example)
#instance:

#webhooks:
# Webhook notifications:
# Possible values:
# Set value to notifiarr if using notifiarr integration
# Set value to apprise if using apprise integration
# Set value to a valid webhook URL
# Set value to nothing (leave Empty/Blank) to disable
#error: https://ntfy.mydomain.com/qbittorrent
#run_start: notifiarr
#run_end: apprise
#function:
#cross_seed: https://ntfy.mydomain.com/qbittorrent
#recheck: notifiarr
#cat_update: apprise
#tag_update: https://ntfy.mydomain.com/qbittorrent
#rem_unregistered: https://ntfy.mydomain.com/qbittorrent
#tag_tracker_error: https://ntfy.mydomain.com/qbittorrent
#rem_orphaned: https://ntfy.mydomain.com/qbittorrent
#tag_nohardlinks: https://ntfy.mydomain.com/qbittorrent
#share_limits: https://ntfy.mydomain.com/qbittorrent
#cleanup_dirs: https://ntfy.mydomain.com/qbittorrent
#bhd:
# BHD Integration used for checking unregistered torrents
#apikey:
#run_start:
#run_end:
```

### Logs

https://gist.github.com/nothing2obvi/4108d89a4f64ea2f41bdfec2e5bc2569

### Installation

Local

### Version Number

4.1.1

### What branch are you on?

master

**Comment by bakerboy448** [comment]
Trace logs are required. The provided logs have nothing to debug and show absolutely no issues nor anything you described

**Comment by nothing2obvi** [comment]
Thanks for taking a look. Please correct me if these are not trace logs:

https://gist.github.com/nothing2obvi/464f06ea62546045701017d5a8d698e9

Just looking at my "public" share limit torrents, I don't see anything about `min_seeding_time` for each torrent even though `min_seeding_time` is set up in my config.

**Comment by bakerboy448** [comment]
```
  public:
	priority: 9
	exclude_any_tags:
	  - something
	  - another-thing
	  - again-another-thing
	max_ratio: 1
	min_seeding_time: 60
	max_seeding_time: 70
	resume_torrent_after_change: True
	cleanup: False
```

you've told qbit_manage to seed for at least 60 seconds and a ratio of 1 OR seed for at least 70 seconds -no bug appears to exist

trace logs indicate this seems to be being applied just fine.

also room to improve the logging an add min seed time, https://github.com/StuffAnThings/qbit_manage/blob/46518577f956fcbf2b18d36c4e9c6d3908b01a95/modules/core/share_limits.py#L196-L219

**Comment by nothing2obvi** [comment]
Wow that's set for 60 seconds and not 60 minutes? I greatly misunderstood, thank you so much.

Yes, adding `min_seeding_time` to logs would be great.

Thanks again!

**Comment by bakerboy448** [comment]
Reopening for the logging cc @bobokun

**Comment by bobokun** [comment]
> Wow that's set for 60 seconds and not 60 minutes?
No, this should be in minutes, not seconds. I think what @bakerboy448 meant was 
> you've told qbit_manage to seed for at least 60 minutes and a ratio of 1 OR seed for at least 70 minutes

**Comment by nothing2obvi** [comment]
@bakerboy448 

So if it's minutes, then `min_seeding_time` really isn't working for me.

**Comment by bakerboy448** [comment]
No it's working accurately.
The values are in second. Thus 60 is 60 seconds
3600 is 1 hour / 60 minutes

**Comment by bakerboy448** [comment]
Trace logging resolved https://github.com/StuffAnThings/qbit_manage/commit/d5aaa31e4057747524c2c9708fd99d78cc7c1cf4

**Comment by nothing2obvi** [comment]
Okay, if it's seconds then I think I should be good.

**Comment by nothing2obvi** [comment]
So I changed my config to account for seconds, not minutes, and my `public` share-limit torrents still aren't seeding. Even the newly added ones since the change.

```
  public:
    priority: 9
    exclude_any_tags:
      - bunch-of-tags-here
    max_ratio: 1
    min_seeding_time: 3600
    max_seeding_time: 4200
    resume_torrent_after_change: True
    cleanup: False
```

![Screen Shot 2024-05-01 at 7 29 00 PM](https://github.com/StuffAnThings/qbit_manage/assets/85228035/17f92a2f-8a6f-4a8e-a287-09a3e1511d09)


https://gist.github.com/nothing2obvi/9ec3106c3b464fad98c7f0b559dd23e9

I've noticed that my torrents from private trackers are all seeding, but those from public trackers aren't. Does that have anything to do with it?

**Comment by bakerboy448** [comment]
Any chance you could try develop which has slightly improved logging on the comparisons

**Comment by nothing2obvi** [comment]
Sorry, I would definitely do it but I'm not sure how to. I'm running a local install. When I go to the develop branch and select `Clone` it tells me the URL is `https://github.com/StuffAnThings/qbit_manage.git` which I don't think points to the develop branch. Unless I'm wrong about that.

**Comment by bakerboy448** [comment]
You simple pull and checkout the develop branch.

**Comment by nothing2obvi** [comment]
Got it to work.

https://gist.github.com/nothing2obvi/cbd215529de9ad9b59dc587f767016de

**Comment by bobokun** [comment]
It's hard to tell because in the screenshot you're hiding the torrent name. But I do see min seed time is working, taking this example:

```
| Torrent: [Bitsearch.to] Prometheus.2012.2160p.UHD.BDRemux.DTS-HD.MA.7.1.HDR.DoVi.Hybrid.P8.by.DVT.mkv [Hash: 344d031905ac72f54d0bcd35d8ae2f532e5caf16] |
| Torrent Category: 4k-movies-radarr                                                                 |
| Torrent Tags: other                                                                                |
| Grouping: public                                                                                   |
| Config Max Ratio vs Torrent Max Ratio:1.0 vs 1                                                     |
| check_max_ratio: False                                                                             |
| Config Max Seeding Time vs Torrent Max Seeding Time: 4200 vs 60                                    |
| Config Max Seeding Time vs Torrent Current Seeding Time: (4200 vs 0.23333333333333334) 2 days, 22:00:00 vs 0:00:14 |
| Config Min Seeding Time vs Torrent Current Seeding Time: (3600 vs 0.23333333333333334) 2 days, 12:00:00 vs 0:00:14 |
| Config Min Num Seeds vs Torrent Num Seeds: 0 vs 3                                                  |
| check_max_seeding_time: True                                                                       |
| Config Limit Upload Speed vs Torrent Limit Upload Speed: -1 vs 488                                 |
| check_limit_upload_speed: True                                                                     |
| hash_not_prev_checked: True                                                                        |
| share_limits_not_yet_tagged: True                                                                  |
|    Torrent Name: [Bitsearch.to] Prometheus.2012.2160p.UHD.BDRemux.DTS-HD.MA.7.1.HDR.DoVi.Hybrid.P8.by.DVT.mkv |
|         Tracker: udp://tracker2.dler.com:80                                                        |
|         Min seed time not met: 0:00:14 <= 2 days, 12:00:00. Removing Share Limits so qBittorrent can continue seeding. |
|         Adding Tag: nmst                                                                           |
|    Torrent Name: [Bitsearch.to] Prometheus.2012.2160p.UHD.BDRemux.DTS-HD.MA.7.1.HDR.DoVi.Hybrid.P8.by.DVT.mkv |
|         Tracker: udp://tracker2.dler.com:80                                                        |
|         Added Tag: sl_9.public                                                                     |
|     Share Limit: Max Ratio = 1.0, Max Seed Time = 4200 min                                         |
```

It shows that `Min seed time not met: 0:00:14 <= 2 days, 12:00:00. Removing Share Limits so qBittorrent can continue seeding.` Is that torrent seeding? If not can you right click the torrent in the webUI and go to `Limit Share Ratio` and screenshot that? Can you also try highlighting these "completed" torrents and choose resume to see if they start seeding?

**Comment by bobokun** [comment]
Can you also check your qbitorrent settings go to the BitTorrent tab? Screenshot the Seeding Limits section 
![image](https://github.com/StuffAnThings/qbit_manage/assets/12660469/a6bfb847-82b1-4c45-8d62-b1110521d14a)

**Comment by bobokun** [comment]
Can you right click one of the torrents that are completed and check what the share limits qbittorrent has set on them? Does it match with what's in your qbm config?

**Comment by nothing2obvi** [comment]
I'm not quite sure how to do that -- I right clicked on a torrent, clicked on `Limit share ratio` and it gave me this:

![Screen Shot 2024-05-02 at 5 25 15 PM](https://github.com/StuffAnThings/qbit_manage/assets/85228035/97cbb1bc-1f04-4e7d-a6d5-0a46401d7776)

Also, as an aside, the numbers in the config really are in minutes then. I was being told it was seconds, if you scroll above.

**Comment by nothing2obvi** [comment]
Upgraded to qBittorrent 4.5.5 and it made no difference.

```
| Torrent: Will.Trent.S02E07.1080p.WEB.h264-ELEANOR[TGx] [Hash: 59742ef3eed0d594141a96a9ee0adae68c8a6015] |
| Torrent Category: tv-sonarr                                                                        |
| Torrent Tags: nmst, other, sl_9.public                                                             |
| Grouping: public                                                                                   |
| Config Max Ratio vs Torrent Max Ratio:1.0 vs -1                                                    |
| check_max_ratio: True                                                                              |
| Config Max Seeding Time vs Torrent Max Seeding Time: 70 vs -1                                      |
| Config Max Seeding Time vs Torrent Current Seeding Time: (70 vs 3.066666666666667) 1:10:00 vs 0:03:04 |
| Config Min Seeding Time vs Torrent Current Seeding Time: (60 vs 3.066666666666667) 1:00:00 vs 0:03:04 |
| Config Min Num Seeds vs Torrent Num Seeds: 0 vs 0                                                  |
| check_max_seeding_time: True                                                                       |
| Config Limit Upload Speed vs Torrent Limit Upload Speed: -1 vs -1                                  |
| check_limit_upload_speed: False                                                                    |
| hash_not_prev_checked: True                                                                        |
| share_limits_not_yet_tagged: False 
```

![Screen Shot 2024-05-03 at 2 41 29 PM](https://github.com/StuffAnThings/qbit_manage/assets/85228035/9d2d2cb5-a377-4be9-b9c5-3697d5634d66)

**Comment by bakerboy448** [comment]
| Config Max Seeding Time vs Torrent Current Seeding Time: (70 vs 3.066666666666667) 1:10:00 vs 0:03:04 |
| Config Min Seeding Time vs Torrent Current Seeding Time: (60 vs 3.066666666666667) 1:00:00 vs 0:03:04 |

Seeding for 3 minutes and not a goal of 70 minutes
Seeding for 3 minutes and not a minimum time of 60 minutes

what's the issue?

also yeah as bobo mentioned, minutes not sconds sorry

**Comment by bobokun** [comment]
I think this is most likely a config issue since if min_seeding_time wasn't working it would be a widespread issue affecting a lot more users. I'm closing this issue, if you need additional support please hop onto the discord as it's easier to debug with you there instead of through Github.

**Comment by nothing2obvi** [comment]
Thanks to both of you and for your help, @bobokun on Discord. I finally solved it and thought I'd share. I noticed that the share limits were working for my downloads from private trackers, but not public trackers. I found a Reddit thread and someone asked if Ultra.cc (my seedbox) supports public trackers. Apparently they do not, and they have scripts that will stop seeding from public trackers once they are done downloading. That's why I could never meet the `min_seeding_time` for public trackers.

https://imgbox.com/3qp2mzfT

Thanks again!
